test/command/test_unpacker.rb

```diff
@@ -0,0 +1,350 @@
+require_relative '../helper'
+
+require 'oj'
+require 'flexmock/test_unit'
+
+require 'fluent/command/unpacker'
+require 'fluent/event'
+
+class TestFluentUnpacker < ::Test::Unit::TestCase
+  module ::UnpackerCommand
+    class Dummy < Base
+      def call; end
+    end
+  end
+
+  def suppress_stdout
+    out = StringIO.new
+    $stdout = out
+    yield
+  ensure
+    $stdout = STDOUT
+  end
+
+  sub_test_case 'call' do
+    data(
+      empty: [],
+      invalid: %w(invalid packed.log),
+    )
+    test 'should fail when invalid command' do |argv|
+      fu = FluentUnpacker.new(argv)
+
+      assert_raise(SystemExit) do
+        suppress_stdout { fu.call }
+      end
+    end
+
+    data(
+      cat: %w(cat packed.log),
+      head: %w(head packed.log),
+      formats: %w(formats packed.log)
+    )
+    test 'should succeed when valid command' do |argv|
+      fu = FluentUnpacker.new(argv)
+
+      flexstub(::UnpackerCommand) do |command|
+        command.should_receive(:const_get).once.and_return(::UnpackerCommand::Dummy)
+        assert_nothing_raised do
+          fu.call
+        end
+      end
+    end
+  end
+end
+
+class TestBaseCommand < ::Test::Unit::TestCase
+  TMP_DIR = File.expand_path(File.dirname(__FILE__) + "/../tmp/command/unpacker#{ENV['TEST_ENV_NUMBER']}")
+
+  def create_message_packed_file(path, times = [event_time], records = [{ 'message' => 'dummy' }])
+    es = Fluent::MultiEventStream.new(times, records)
+    v = es.to_msgpack_stream
+    File.open("#{TMP_DIR}/#{path}", 'w') do |f|
+      f.print(v)
```

Comment:
Generally speaking, it's better practice to use `IO#write` to output binary data without any modification. `IO#print` may be affected by separator character (with multiple arguments).
It's just comment. No need to update here.